### DISCUSSION TITLE
Colorised headers on Windows

### DISCUSSION THREAD

**Discussion by gvanem** [curl/curl#11770] [Ideas]
The ANSI ESC-codes in `src/tool_cb_hdr.c` to enable colorised header output is disabled for `WIN32`:
```c
#ifdef WIN32
#define BOLD
#define BOLDOFF
#else
#define BOLD "\x1b[1m"
...
```

But with a little patch:
```diff
--- a/src/tool_cb_hdr.c 2023-08-30 14:52:41
+++ b/src/tool_cb_hdr.c 2023-08-30 14:58:49
@@ -41,7 +41,7 @@

 static char *parse_filename(const char *ptr, size_t len);

-#ifdef WIN32
+#if defined(WIN32) && !defined(USE_VT_CODES)
 #define BOLD
 #define BOLDOFF
```

a `--head` option gives colours on Windows too:
![curl-colour-headers](https://github.com/curl/curl/assets/945271/58449a18-98ca-45a0-99a9-af0e4d6c3d4a)

Except that ESC codes for `Location:` is unsupported in the Windows Console.

To use ESC-codes in *Windows Console*  this should be TRUE:
```c
DWORD mode;
bool have_vts = (GetConsoleMode(GetStdHandle(STD_OUTPUT_HANDLE), &mode) &&
                   (mode & ENABLE_VIRTUAL_TERMINAL_PROCESSING));
```
Otherwise colourised headers should be disabled at *run-time*.

I'm just fishing to hear if there's some interest for colourised output on Windows.

**Comment by dfandrich** [comment]
I think this is worthwhile having for feature parity with Linux, although I'm not a Windows user. The `have_vts` snippet looks analogous to the `isatty` flag; maybe that's how it should be set on Windows? Or does the existing code calling `isatty()` already do the right thing on Windows? Are you willing to turn this into a PR?

**Reply by gvanem** [reply]
>  the isatty flag; 

This: `if(hdrcbdata->global->isatty` ? 

>  Or does the existing code calling isatty() already do the right thing on Windows?

Yes, since: `curl.exe --head --location http://curl.se | tee.exe foo` has no colours.

>  Are you willing to turn this into a PR?

I think not.

*PS*. I think the options `--head` and `--header` are confusingly similar.
